### PR TITLE
[Feature] Add TTA transform

### PR DESCRIPTION
**Pull request by HAOCHENYE**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
Transforms used by test time augmentation.

## Modification
Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)
Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)
```python
transforms = [
    [dict(type='Resize', scale=(1333, 800), keep_ratio=True),
     dict(type='Resize', scale=(1333, 800), keep_ratio=True)], 
    [dict(type='RandomFlip', prob=1.),
     dict(type='RandomFlip', prob=0.)], 
    [dict(type='Normalize', mean=(0, 0, 0), std=(1, 1, 1))]]

tta_transform = TestTimeAug(transforms)
```

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by mzr1996** [comment]
Maybe use a cartesian product to implement the subroutines

```python
from itertools import product

@TRANSFORMS.register_module()
class TestTimeAug(BaseTransform):

    def __init__(self, transforms: list):
        self.subroutines = [Compose(subroutine) for subroutine in product(*transfroms)]

    def transform(self, results: dict) -> dict:
        results_list = []  # type: ignore
        for subroutine in self.subroutines:
            routine_results = subroutine(deepcopy(results))
            results_list.append(routine_results)

        aug_data_dict = {
            key: [item[key] for item in results]
            for key in results_list[0]
        }
        return aug_data_dict
```

File: mmcv/transforms/processing.py [hunk 1]
```diff
@@ -868,6 +869,117 @@ def __repr__(self) -> str:
...
+
+    For the above configuration, there are four combinations of resize
+    and flip:
+
+    - Resize to (1333, 400) + no flip
+    - Resize to (1333, 400) + flip
+    - Resize to (1333, 800) + no flip
+    - resize to (1333, 800) + flip
+
+    After that, results are wrapped into lists of the same length as followed:
```
Comment (ZwwWayne) [inline mmcv/transforms/processing.py:902]: as below or as the following, there is no  usages of `as followed`

File: mmcv/transforms/processing.py [hunk 2]
```diff
@@ -868,6 +869,117 @@ def __repr__(self) -> str:
...
+
+    .. code-block::
+
+        dict(
+            inputs=[...],
+            data_samples=[...]
+        )
+
+    Where the length of ``inputs`` and ``data_samples`` are both 4.
```
Comment (ZwwWayne) [inline mmcv/transforms/processing.py:911]: Where the -> The length

**Review by mzr1996 — APPROVED** [review]

File: mmcv/transforms/processing.py [hunk 3]
```diff
@@ -870,6 +872,130 @@ def __repr__(self) -> str:
...
+        2 ``Resize``, 2 ``RandomFlip`` and 1 ``PackDetInputs``.
+        The data sampled from dataset will be processed as follows:
+
+        1. Data will be processed by 2 ``Resize`` and return a list
+           of 2 results.
+        2. Each result in list will be further passed to 2
+           ``RandomFlip``, and aggregates into a list of 4 results.
+        3. Each result will be processed by ``PackDetInputs``, and
+           return a list of dict.
+        4. Aggregates the same fields of results, and finally return
```
Comment (zhouzaida) [inline mmcv/transforms/processing.py:962]:
```suggestion
        4. Aggregates the same fields of results, and finally returns
```

File: tests/test_transforms/test_transforms_processing.py [hunk 4]
```diff
@@ -900,3 +900,89 @@ def test_transform(self):
                                       resize_type='Resize',
                                       keep_ratio=True)
             results_update = TRANSFORMS.transform(copy.deepcopy(results))
+
+
+class TestTestTimeAug:
+
+    def test_init(self):
+        subroutines = [[
+            dict(type='Resize', scale=(1333, 800), keep_ratio=True),
```
Comment (zhouzaida) [inline tests/test_transforms/test_transforms_processing.py:909]:
```suggestion
            dict(type='Resize', scale=(1333, 400), keep_ratio=True),
```

**Pull request merged**